flowey/flowey_lib_hvlite/src/build_and_test_vmgs_lib.rs

```diff
@@ -12,17 +12,24 @@ use flowey::node::prelude::*;
 use flowey_lib_common::run_cargo_build::CargoCrateType;
 
 #[derive(Serialize, Deserialize)]
+#[serde(untagged)]
 pub enum VmgsLibOutput {
     LinuxDynamicLib {
+        #[serde(rename = "libvmgs_lib.so")]
```

Comment:
leaning on serde here is very clever! It'll be interesting to see how this scales to more complex (and nested) structures in the future.

also, like we talked about offline - weaning flowey off relying on serde's derive macros directly, and onto something a bit more restricted / flowey-controlled will be really important to avoid introducing accidental footguns in the future